src/ngx_lua_resty_lmdb_module.c

```diff
@@ -338,6 +456,133 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
 }
 
 
+static ngx_int_t
+ngx_lua_resty_lmdb_close_file(ngx_cycle_t *cycle,
+                              ngx_lua_resty_lmdb_conf_t *lcf)
+{
+    mdb_txn_abort(lcf->ro_txn);
+    mdb_env_close(lcf->env);
+
+    lcf->ro_txn = NULL;
+    lcf->env = NULL;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_lua_resty_lmdb_verify_file_status(ngx_cycle_t *cycle,
+                                      ngx_lua_resty_lmdb_conf_t *lcf)
+{
+    ngx_core_conf_t  *ccf;
+    ngx_file_info_t   fi;
+
+    u_char            name_buf[NGX_MAX_PATH];
+    ngx_str_t        *names = ngx_lua_resty_lmdb_file_names;
+    ngx_str_t        *name;
+
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+
+    if (ccf->user == (ngx_uid_t) NGX_CONF_UNSET_UINT) {
+        return NGX_OK;
+    }
+
+    /* check directory */
+
+    ngx_snprintf(name_buf, NGX_MAX_PATH,
+                 "%V%Z", &lcf->env_path->name);
+
+    if (ngx_file_info(name_buf, &fi) == NGX_FILE_ERROR) {
+        ngx_log_error(NGX_LOG_CRIT, cycle->log, ngx_errno,
+                      ngx_file_info_n " \"%s\" failed", name_buf);
+        return NGX_ERROR;
+    }
+
+    if (fi.st_uid != ccf->user) {
+        if (chown((const char *) name_buf, ccf->user, -1) == -1) {
+            ngx_log_error(NGX_LOG_CRIT, cycle->log, ngx_errno,
+                          "chown(\"%s\", %d) failed",
+                          name_buf, ccf->user);
+            return NGX_ERROR;
```

Comment:
Maybe make this fallible. I saw Nginx also allows `chown` to fail:

https://lxr.nginx.org/source/xref/nginx/src/core/ngx_cycle.c?r=7744%3Af18db38a9826#1249

Reply:
done.